src/test/java/it/gov/pagopa/payhub/auth/service/a2a/legacy/ValidateJWTLegacyServiceTest.java

```diff
@@ -0,0 +1,120 @@
+package it.gov.pagopa.payhub.auth.service.a2a.legacy;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.interfaces.Claim;
+import it.gov.pagopa.payhub.auth.exception.custom.InvalidTokenException;
+import it.gov.pagopa.payhub.auth.utils.JWTValidator;
+import it.gov.pagopa.payhub.auth.utils.JWTValidatorUtils;
+import org.apache.commons.lang3.tuple.Pair;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.security.KeyPair;
+import java.security.PublicKey;
+import java.time.Instant;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
+
+
+@ExtendWith(MockitoExtension.class)
+class ValidateJWTLegacyServiceTest {
+
+	@Mock
+	private A2AClientLegacyPropConfig a2AClientLegacyPropConfig;
+	@Mock
+	private JWTValidator jwtValidatorMock;
+	private ValidateJWTLegacyService service;
+	private KeyPair keyPair;
+
+
+	@BeforeEach
+	void setup() throws Exception {
+		service = new ValidateJWTLegacyService(a2AClientLegacyPropConfig, jwtValidatorMock);
+		keyPair = JWTValidatorUtils.generateKeyPair();
+	}
+
+	@Test
+	void GivenValidTokenThenOk() {
+		String appName = "A2A-IPA_TEST_1";
+		PublicKey publicKey = keyPair.getPublic();
+		String token = JWTValidatorUtils.generateLegacyToken(keyPair, "a2a", Instant.now(), Instant.now().plusSeconds(3_600_000L), "jti");
+		when(a2AClientLegacyPropConfig.getPublicKeysAsMap()).thenReturn(Map.of(appName, publicKey));
```

Comment:
fill the map with 2 key, the first not valid (whose invocation should throw the invalid exception) and the second the valid:

the behavior should be to have an OK, because there is an entry which will validate it